segway/gmtk/params.py

```diff
@@ -150,6 +186,23 @@ def uniform_from_shape(cls, *shape: int,
         return DenseCPT(values, keep_shape=True)
 
 
+class DirichletTable(MultiDimArray):
+    """
+    A single DirichletTable object.
+    """
+    kind = OBJ_KIND_DIRICHLETTAB
+
+    def get_header_info(self) -> str:
+        """
+        Return number of parents, cardinality line, for header in
+        input.master section.
+        """
+        line = [str(len(self.shape))]  # number of parents
```

Comment:
This capability seems to be needed both here and MX at least. Can you make a method that does:

```
def make_cardinality_line(shape: ndarray):
    num_parents = len(shape)
    cardinalities = shape

   return " ".join(map(str, [num_parents, *cardinalities]))
```

And then you can use it here and also in `MX`.

Reply:
Added and used to implement this method.